### ISSUE TITLE
Mobile resolution and page zoom will cause content to be hidden or a horizontal scroll to show

### ISSUE THREAD

**Issue by karlgroves** [WordPress/gutenberg#15346]
Mobile resolution and page zoom will cause content to be hidden or a horizontal scroll to show

-   **Severity:** High
-   **Affected Populations:**
    -   Low-Vision
    -   Motor Impaired
    -   Cognitively Impaired
-   **Platform(s):**
    -   All / Universal
-   **Components affected:**
    -   Global

#### Issue description

The Editor Top Bar remains sticky to the top of the page even as users
zoom up to 400%. Two problems occur from this:

-   Depending on viewport resolution, the last button or two may go
    offscreen to the right (this is due to flexbox). Keyboard users can
    tab to these buttons, however touchscreen users and speech
    recognition users cannot reach these buttons, as there is no
    scrollbar.
-   Depending on viewport height, users attempting to write content find
    a large part of their viewport is blocked by the sticky bar and
    partially covers typed text.

When the pages are zoomed, or in smaller windows, there is a horizontal
scrollbar. In IE 11 it is shown as an inner scrollbar that overlaps or
has a limited width so cannot always be used. On mobile the finger
swipes work as expected. This also affects wide panels that do not wrap
into new lines but overlap or go off the page.

The ability to zoom content may be essential for users with low-vision,
and may be helpful for users who have a cognitive disability.

##### Issue Code

```
    .edit-post-header {


        ...


        display: flex;


        flex-direction: row;


        align-items: stretch;


        justify-content: space-between;


        position: -webkit-sticky;


        position: sticky;


    }





    /* media all */


    .edit-post-layout__content {


        position: relative;


        display: flex;


        min-height: 100%;


        flex-direction: column;


        padding-bottom: 50vh;


        overflow-y: auto;


        -webkit-overflow-scrolling: touch;


    }
```

#### Remediation Guidance

For the post header set the `position` attribute to `block` instead
of `sticky` when a media query detects the viewport is small (20-30
em). This will hit zoom levels under 400% and allow users who zoom quite
a lot to simply scroll past the bar.

Use the same media query to adjust the `flex` settings to allow
wrapping; or implement a different design solution to the same effect.
Currently, flex elements' alignment does not account for viewport
edges, and allows content to go off screen without creating scrollbars;
this effectively hides the overflowing content.

In general, when the screen is zoomed-in or in a low resolution, do not
rely on horizontal scrolling: fit items into the screen by wrapping the
content onto new lines when the resolution is smaller than the width of
the content.

Allow links styled as buttons to wrap their text. Actual HTML buttons
wrap naturally but have a set `line-height` of `1`, making their
text difficult to read when wrapping. Set this to the same
`line-height` as link-buttons (`1.66`).

Allow tables to be a width where no individual words need to be broken,
by setting the wrapper of the table to have `overflow-x: scroll` so
the table does not cause the page itself to have a horizontal scrollbar.
Remove the `word-break` rule used in tables.

#### Relevant standards

-   [1.4.10 Reflow (Level
    AA)](https://www.w3.org/TR/WCAG20/#visual-audio-contrast-without-color)

**Note:** This issue may be a duplicate with other existing accessibility-related bugs in this project. This issue comes from the Gutenberg accessibility audit, performed by [Tenon](https://www.tenon.io) and funded by [WP Campus](https://wpcampus.org/). This issue is GUT-4 in Tenon's report

**Comment by melchoyce** [comment]
Screenshot from full report: 
![image](https://user-images.githubusercontent.com/2846578/57185760-1f38cd00-6e86-11e9-880e-2e881a597a69.png)

**Comment by mapk** [comment]
The solution provided in the issue appears simple enough. I don't think it requires any design work, just need a fix.

**Comment by afercia** [comment]
Discussed during today's accessibility bug scrub, agreed all the recommendations here are valuable and actionable. 

Particularly interesting the suggestion to “unstick” the top bar at very high zoom levels.

**Comment by SchneiderSam** [comment]
Am I getting this right? If this is fixed, then the table-Block will be Responsive?

That would be brilliant!

**Comment by tellthemachines** [comment]
I have just made a PR addressing the top bar and table issues. 
Still to do:
* Links styled as buttons should wrap
* Close button on sticky warning banners is hidden from view at high zoom levels

The links-as-buttons issue doesn't seem to be an overall one but specific instances such as "Change Permalinks" on the title block need fixing. I'm looking at a solution for that and will do another PR when ready.

<img width="1195" alt="Screen Shot 2019-06-24 at 9 35 59 am" src="https://user-images.githubusercontent.com/8096000/59984277-a4ad5200-966b-11e9-969d-5b1bcfd0f698.png">

<img width="1065" alt="Screen Shot 2019-06-24 at 9 34 35 am" src="https://user-images.githubusercontent.com/8096000/59984282-ad058d00-966b-11e9-94fd-cb34f0594b53.png">

**Comment by kjellr** [comment]
Opening this back up for those two items [noted above](https://github.com/WordPress/gutenberg/issues/15346#issuecomment-504802754).
